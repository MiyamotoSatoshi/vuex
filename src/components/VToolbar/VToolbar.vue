<script>
  import Themeable from '../../mixins/themeable'

  export default {
    name: 'v-toolbar',

    mixins: [Themeable],

    data: () => ({
      isExtended: false,
      isScrolling: false,
      marginTop: 0,
      previousScroll: null,
      target: null
    }),

    props: {
      absolute: Boolean,
      card: Boolean,
      dense: Boolean,
      extended: Boolean,
      fixed: Boolean,
      flat: Boolean,
      floating: Boolean,
      prominent: Boolean,
      scrollOffScreen: Boolean,
      scrollTarget: String,
      scrollThreshold: {
        type: Number,
        default: 100
      }
    },

    computed: {
      classes () {
        return {
          'toolbar': true,
          'elevation-0': this.flat,
          'toolbar--absolute': this.absolute,
          'toolbar--card': this.card,
          'toolbar--dense': this.dense,
          'toolbar--fixed': this.fixed,
          'toolbar--floating': this.floating,
          'toolbar--prominent': this.prominent,
          'toolbar--extended': this.isExtended,
          'theme--dark': this.dark,
          'theme--light': this.light
        }
      },
      styles () {
        return {
          marginTop: this.marginTop
        }
      }
    },

    watch: {
      isScrolling (val) {
        if (!val) this.marginTop = 0
        else (this.marginTop = `${-this.$refs.content.clientHeight - 6}px`)
      }
    },

    methods: {
      onScroll () {
        if (typeof window === 'undefined') return

        if (!this.target) {
          this.target = this.scrollTarget
            ? document.querySelector(this.scrollTarget)
            : window
        }

        const currentScroll = this.scrollTarget
          ? this.target.scrollTop
          : this.target.pageYOffset || document.documentElement.scrollTop

        if (currentScroll < this.scrollThreshold) return

        if (this.previousScroll === null) {
          this.previousScroll = currentScroll
        }

        this.isScrolling = this.previousScroll < currentScroll

        this.previousScroll = currentScroll
      }
    },

    render (h) {
      this.isExtended = this.extended || !!this.$slots.extension
      const children = []
      const data = {
        'class': this.classes,
        style: this.styles,
        on: this.$listeners
      }

      if (this.scrollOffScreen) {
        data.directives = [{
          name: 'scroll',
          value: {
            callback: this.onScroll,
            target: this.scrollTarget
          }
        }]
      }

      children.push(h('div', {
        'class': 'toolbar__content',
        ref: 'content'
      }, this.$slots.default))

      if (this.isExtended) {
        children.push(h('div', {
          'class': 'toolbar__extension'
        }, this.$slots.extension))
      }

      return h('nav', data, children)
    }
  }
</script>

<style lang="stylus" src="../../stylus/components/_toolbar.styl"></style>
