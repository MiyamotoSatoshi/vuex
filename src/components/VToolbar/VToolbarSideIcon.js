import VBtn from '../../components/VBtn'
import VIcon from '../../components/VIcon'

/* @vue/component */
export default {
  name: 'v-toolbar-side-icon',

  functional: true,

  render (h, { slots, listeners, props, data }) {
    const classes = data.staticClass
      ? `${data.staticClass} v-toolbar__side-icon`
      : 'v-toolbar__side-icon'

    const d = Object.assign(data, {
      staticClass: classes,
      props: Object.assign(props, {
        icon: true
      }),
      on: listeners
    })

    const defaultSlot = slots().default

    return h(VBtn, d, defaultSlot || [h(VIcon, '$vuetify.icons.menu')])
  }
}
