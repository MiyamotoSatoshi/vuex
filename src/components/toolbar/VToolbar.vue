<script>
  import Themeable from '../../mixins/themeable'

  export default {
    name: 'v-toolbar',

    mixins: [Themeable],

    data: () => ({
      isExtended: false
    }),

    props: {
      absolute: Boolean,
      card: Boolean,
      dense: Boolean,
      extended: Boolean,
      fixed: Boolean,
      flat: Boolean,
      floating: Boolean,
      prominent: Boolean
    },

    computed: {
      classes () {
        return {
          'toolbar': true,
          'elevation-0': this.flat,
          'toolbar--absolute': this.absolute,
          'toolbar--card': this.card,
          'toolbar--dense': this.dense,
          'toolbar--fixed': this.fixed,
          'toolbar--floating': this.floating,
          'toolbar--prominent': this.prominent,
          'toolbar--extended': this.isExtended,
          'theme--dark': this.dark,
          'theme--light': this.light
        }
      }
    },

    render (h) {
      this.isExtended = this.extended || !!this.$slots.extension
      const children = []
      const data = {
        'class': this.classes,
        on: this.$listeners
      }

      children.push(h('div', {
        'class': 'toolbar__content'
      }, this.$slots.default))

      if (this.isExtended) {
        children.push(h('div', {
          'class': 'toolbar__extension'
        }, this.$slots.extension))
      }

      return h('nav', data, children)
    }
  }
</script>

<style lang="stylus" src="../../stylus/components/_toolbar.styl"></style>
