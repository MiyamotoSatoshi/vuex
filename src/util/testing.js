import Vue from 'vue'
import load from '~util/load'
import { mount, shallow } from 'avoriaz'

export function test(name, cb) {
  Vue.prototype.$vuetify = { load }

  beforeEach(() => {
    rafPolyfill(window)
  })

  describe(name, () => cb({
    functionalContext,
    mount,
    shallow
  }))
}

export function functionalContext(context = {}) {
  return {
    context: Object.assign({
      data: {},
      props: {}
    }, context)
  }
}


//requestAnimationFrame polyfill | Milos Djakonovic ( @Miloshio ) | MIT | https://github.com/milosdjakonovic/requestAnimationFrame-polyfill
export function rafPolyfill(w) {
  /**
   *
   * How many times should polyfill call
   * update callback? By canon, it should
   * be 60 times per second, so that ideal
   * framerate 60fps could be reached.
   *
   * However, even native implementations
   * of requestAnimationFrame often cannot
   * do 60fps, but, unlike any polyfill,
   * they can synchronise achievable fps
   * rate with screen refresh rate.
   *
   * So, leave this value 1000/60 unless
   * you target specific browser on spec
   * ific device that is going to work
   * better with custom value. I think
   * that this is the longest comment I've
   * written on single variable so far.
  **/
  var FRAME_RATE_INTERVAL = 1000/60,

  /**
   * All queued callbacks in given cycle.
  **/
  allCallbacks = [],

  executeAllScheduled = false,

  shouldCheckCancelRaf = false,

  /**
   * Callbacks queued for cancellation.
  **/
  callbacksForCancellation = [],

  /**
   * Should callback be cancelled?
   * @param cb - callback
  **/
  isToBeCancelled = function(cb){
    for(var i=0;i<callbacksForCancellation.length;i++){
      if(callbacksForCancellation[i] === cb ){
        callbacksForCancellation.splice(i,1);
        return true;
      }
    }
  },



  /**
   *
   * Executes all (surprise) callbacks in
   * and removes them from callback queue.
   *
  **/
  executeAll = function(){
    executeAllScheduled = false;
    var _allCallbacks = allCallbacks;
    allCallbacks = [];
    for(var i=0;i<_allCallbacks.length;i++){
      if(shouldCheckCancelRaf===true){
        if (isToBeCancelled(_allCallbacks[i])){
          shouldCheckCancelRaf = false;
          return;
        }
      }
      _allCallbacks[i].apply(w, [ new Date().getTime() ] );
    }
  },

  /**
   *
   * requestAnimationFrame polyfill
   * @param callback - callback to be queued & executed | executed
   * @return callback
   *
  **/
  raf = function(callback){
    allCallbacks.push(callback);
    if(executeAllScheduled===false){
      w.setTimeout(executeAll, FRAME_RATE_INTERVAL);
      executeAllScheduled = true;
    }
    return callback;
  },

  /**
   *
   * Cancels raf.
  **/
  cancelRaf = function(callback){
    callbacksForCancellation.push(callback);
    shouldCheckCancelRaf = true;
  },


  //https://gist.github.com/paulirish/1579671
  vendors = ['ms', 'moz', 'webkit', 'o'];

    for(var x = 0; x < vendors.length && !w.requestAnimationFrame; ++x) {
        w.requestAnimationFrame = w[vendors[x]+'RequestAnimationFrame'];
        w.cancelAnimationFrame = w[vendors[x]+'CancelAnimationFrame']
        || w[vendors[x]+'CancelRequestAnimationFrame'];
    }

  if (!w.requestAnimationFrame) w.requestAnimationFrame = raf;
  if (!w.cancelAnimationFrame)  w.cancelAnimationFrame  = cancelRaf;

}
