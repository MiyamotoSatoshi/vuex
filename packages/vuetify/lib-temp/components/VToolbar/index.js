import { createSimpleFunctional } from '../../util/helpers';
import VToolbar from './VToolbar';
import VToolbarSideIcon from './VToolbarSideIcon';
const VToolbarTitle = createSimpleFunctional('v-toolbar__title');
const VToolbarItems = createSimpleFunctional('v-toolbar__items');
export { VToolbar, VToolbarSideIcon, VToolbarTitle, VToolbarItems };
export default {
    $_vuetify_subcomponents: {
        VToolbar,
        VToolbarItems,
        VToolbarTitle,
        VToolbarSideIcon
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WVG9vbGJhci9pbmRleC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsc0JBQXNCLEVBQUUsTUFBTSxvQkFBb0IsQ0FBQTtBQUUzRCxPQUFPLFFBQVEsTUFBTSxZQUFZLENBQUE7QUFDakMsT0FBTyxnQkFBZ0IsTUFBTSxvQkFBb0IsQ0FBQTtBQUVqRCxNQUFNLGFBQWEsR0FBRyxzQkFBc0IsQ0FBQyxrQkFBa0IsQ0FBQyxDQUFBO0FBQ2hFLE1BQU0sYUFBYSxHQUFHLHNCQUFzQixDQUFDLGtCQUFrQixDQUFDLENBQUE7QUFFaEUsT0FBTyxFQUFFLFFBQVEsRUFBRSxnQkFBZ0IsRUFBRSxhQUFhLEVBQUUsYUFBYSxFQUFFLENBQUE7QUFFbkUsZUFBZTtJQUNiLHVCQUF1QixFQUFFO1FBQ3ZCLFFBQVE7UUFDUixhQUFhO1FBQ2IsYUFBYTtRQUNiLGdCQUFnQjtLQUNqQjtDQUNGLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBjcmVhdGVTaW1wbGVGdW5jdGlvbmFsIH0gZnJvbSAnLi4vLi4vdXRpbC9oZWxwZXJzJ1xuXG5pbXBvcnQgVlRvb2xiYXIgZnJvbSAnLi9WVG9vbGJhcidcbmltcG9ydCBWVG9vbGJhclNpZGVJY29uIGZyb20gJy4vVlRvb2xiYXJTaWRlSWNvbidcblxuY29uc3QgVlRvb2xiYXJUaXRsZSA9IGNyZWF0ZVNpbXBsZUZ1bmN0aW9uYWwoJ3YtdG9vbGJhcl9fdGl0bGUnKVxuY29uc3QgVlRvb2xiYXJJdGVtcyA9IGNyZWF0ZVNpbXBsZUZ1bmN0aW9uYWwoJ3YtdG9vbGJhcl9faXRlbXMnKVxuXG5leHBvcnQgeyBWVG9vbGJhciwgVlRvb2xiYXJTaWRlSWNvbiwgVlRvb2xiYXJUaXRsZSwgVlRvb2xiYXJJdGVtcyB9XG5cbmV4cG9ydCBkZWZhdWx0IHtcbiAgJF92dWV0aWZ5X3N1YmNvbXBvbmVudHM6IHtcbiAgICBWVG9vbGJhcixcbiAgICBWVG9vbGJhckl0ZW1zLFxuICAgIFZUb29sYmFyVGl0bGUsXG4gICAgVlRvb2xiYXJTaWRlSWNvblxuICB9XG59XG4iXX0=