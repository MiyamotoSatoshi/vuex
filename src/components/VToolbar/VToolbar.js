// Styles
import '../../stylus/components/_toolbar.styl'

// Mixins
import Applicationable from '../../mixins/applicationable'
import Colorable from '../../mixins/colorable'
import Themeable from '../../mixins/themeable'
import SSRBootable from '../../mixins/ssr-bootable'

// Directives
import Scroll from '../../directives/scroll'
import { deprecate } from '../../util/console'

/* @vue/component */
export default {
  name: 'v-toolbar',

  directives: { Scroll },

  mixins: [
    Applicationable('top', [
      'clippedLeft',
      'clippedRight',
      'computedHeight',
      'invertedScroll',
      'manualScroll'
    ]),
    Colorable,
    SSRBootable,
    Themeable
  ],

  props: {
    card: Boolean,
    clippedLeft: Boolean,
    clippedRight: Boolean,
    dense: Boolean,
    extended: Boolean,
    extensionHeight: {
      type: [Number, String],
      validator: v => !isNaN(parseInt(v))
    },
    flat: Boolean,
    floating: Boolean,
    height: {
      type: [Number, String],
      validator: v => !isNaN(parseInt(v))
    },
    invertedScroll: Boolean,
    manualScroll: Boolean,
    prominent: Boolean,
    scrollOffScreen: Boolean,
    /* @deprecated */
    scrollToolbarOffScreen: Boolean,
    scrollTarget: String,
    scrollThreshold: {
      type: Number,
      default: 300
    },
    tabs: Boolean
  },

  data: () => ({
    activeTimeout: null,
    currentScroll: 0,
    heights: {
      mobileLandscape: 48,
      mobile: 56,
      desktop: 64,
      dense: 48
    },
    isActive: true,
    isExtended: false,
    isScrollingUp: false,
    previousScroll: null,
    previousScrollDirection: null,
    savedScroll: 0,
    target: null
  }),

  computed: {
    canScroll () {
      // TODO: remove
      if (this.scrollToolbarOffScreen) {
        deprecate('scrollToolbarOffScreen', 'scrollOffScreen', this)

        return true
      }

      return this.scrollOffScreen || this.invertedScroll
    },
    computedContentHeight () {
      if (this.height) return parseInt(this.height)
      if (this.dense) return this.heights.dense

      if (this.prominent ||
        this.$vuetify.breakpoint.mdAndUp
      ) return this.heights.desktop

      if (this.$vuetify.breakpoint.width >
        this.$vuetify.breakpoint.height
      ) return this.heights.mobileLandscape

      return this.heights.mobile
    },
    computedExtensionHeight () {
      if (this.tabs) return 48
      if (this.extensionHeight) return parseInt(this.extensionHeight)

      return this.computedContentHeight
    },
    computedHeight () {
      if (!this.isExtended) return this.computedContentHeight

      return this.computedContentHeight + this.computedExtensionHeight
    },
    computedMarginTop () {
      if (!this.app) return 0

      return this.$vuetify.application.bar
    },
    classes () {
      return this.addBackgroundColorClassChecks({
        'v-toolbar': true,
        'elevation-0': this.flat || (!this.isActive &&
          !this.tabs &&
          this.canScroll
        ),
        'v-toolbar--absolute': this.absolute,
        'v-toolbar--card': this.card,
        'v-toolbar--clipped': this.clippedLeft || this.clippedRight,
        'v-toolbar--dense': this.dense,
        'v-toolbar--extended': this.isExtended,
        'v-toolbar--fixed': !this.absolute && (this.app || this.fixed),
        'v-toolbar--floating': this.floating,
        'v-toolbar--prominent': this.prominent,
        'theme--dark': this.dark,
        'theme--light': this.light
      })
    },
    computedPaddingLeft () {
      if (!this.app || this.clippedLeft) return 0

      return this.$vuetify.application.left
    },
    computedPaddingRight () {
      if (!this.app || this.clippedRight) return 0

      return this.$vuetify.application.right
    },
    computedTransform () {
      return !this.isActive
        ? this.canScroll
          ? -this.computedContentHeight
          : -this.computedHeight
        : 0
    },
    currentThreshold () {
      return Math.abs(this.currentScroll - this.savedScroll)
    },
    styles () {
      return {
        marginTop: `${this.computedMarginTop}px`,
        paddingRight: `${this.computedPaddingRight}px`,
        paddingLeft: `${this.computedPaddingLeft}px`,
        transform: `translateY(${this.computedTransform}px)`
      }
    }
  },

  watch: {
    currentThreshold (val) {
      if (this.invertedScroll) {
        return this.isActive = this.currentScroll > this.scrollThreshold
      }

      if (val < this.scrollThreshold ||
        !this.isBooted
      ) return

      this.isActive = this.isScrollingUp
      this.savedScroll = this.currentScroll
    },
    isActive () {
      this.savedScroll = 0
    },
    invertedScroll (val) {
      this.isActive = !val
    },
    manualScroll (val) {
      this.isActive = !val
    },
    isScrollingUp () {
      this.savedScroll = this.savedScroll || this.currentScroll
    }
  },

  created () {
    if (this.invertedScroll ||
      this.manualScroll
    ) this.isActive = false
  },

  mounted () {
    if (this.scrollTarget) {
      this.target = document.querySelector(this.scrollTarget)
    }
  },

  methods: {
    onScroll () {
      if (!this.canScroll ||
        this.manualScroll ||
        typeof window === 'undefined'
      ) return

      const target = this.target || window

      this.currentScroll = this.scrollTarget
        ? target.scrollTop
        : target.pageYOffset || document.documentElement.scrollTop

      this.isScrollingUp = this.currentScroll < this.previousScroll

      this.previousScroll = this.currentScroll
    },
    /**
     * Update the application layout
     *
     * @return {number}
     */
    updateApplication () {
      return this.invertedScroll || this.manualScroll
        ? 0
        : this.computedHeight
    }
  },

  render (h) {
    this.isExtended = this.extended || !!this.$slots.extension

    const children = []
    const data = {
      'class': this.classes,
      style: this.styles,
      on: this.$listeners
    }

    data.directives = [{
      arg: this.scrollTarget,
      name: 'scroll',
      value: this.onScroll
    }]

    children.push(h('div', {
      staticClass: 'v-toolbar__content',
      style: { height: `${this.computedContentHeight}px` },
      ref: 'content'
    }, this.$slots.default))

    if (this.isExtended) {
      children.push(h('div', {
        staticClass: 'v-toolbar__extension',
        style: { height: `${this.computedExtensionHeight}px` }
      }, this.$slots.extension))
    }

    return h('nav', data, children)
  }
}
