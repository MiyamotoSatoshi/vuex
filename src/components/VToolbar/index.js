import {
  createSimpleFunctional
} from '../../util/helpers'

import VToolbar from './VToolbar'
import VToolbarSideIcon from './VToolbarSideIcon'

const VToolbarTitle = createSimpleFunctional('v-toolbar__title')
const VToolbarItems = createSimpleFunctional('v-toolbar__items')

export { VToolbar, VToolbarSideIcon, VToolbarTitle, VToolbarItems }

/* istanbul ignore next */
VToolbar.install = function install (Vue) {
  Vue.component(VToolbar.name, VToolbar)
  Vue.component(VToolbarItems.name, VToolbarItems)
  Vue.component(VToolbarTitle.name, VToolbarTitle)
  Vue.component(VToolbarSideIcon.name, VToolbarSideIcon)
}

export default VToolbar
