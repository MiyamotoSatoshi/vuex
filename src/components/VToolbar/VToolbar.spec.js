import Vue from 'vue'
import { test } from '~util/testing'
import VToolbar from '~components/VToolbar'

const scrollWindow = y => {
  global.pageYOffset = y
  global.dispatchEvent(new Event('scroll'))

  return new Promise(resolve => setTimeout(resolve, 200))
}

test('VToolbar.vue', ({ mount }) => {
  it('should render a colored toolbar', () => {
    const wrapper = mount(VToolbar, {
      propsData: {
        color: 'blue lighten-1'
      }
    })

    expect(wrapper.element.classList).toContain('blue')
    expect(wrapper.element.classList).toContain('lighten-1')
  })

  it('should render an extended toolbar', () => {
    const wrapper = mount(VToolbar, {
      propsData: {
        extended: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render an extended toolbar with specific height', () => {
    const wrapper = mount(VToolbar, {
      propsData: {
        extended: true,
        extensionHeight: 42
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should calculate paddings ', () => {
    const wrapper = mount(VToolbar)

    wrapper.vm.$vuetify.application.left = 42
    wrapper.vm.$vuetify.application.right = 84

    wrapper.setProps({ app: false, clippedLeft: false, clippedRight: false })
    expect(wrapper.vm.computedPaddingLeft).toBe(0)
    expect(wrapper.vm.computedPaddingRight).toBe(0)
    wrapper.setProps({ app: false, clippedLeft: true, clippedRight: true })
    expect(wrapper.vm.computedPaddingLeft).toBe(0)
    expect(wrapper.vm.computedPaddingRight).toBe(0)
    wrapper.setProps({ app: true, clippedLeft: false, clippedRight: false })
    expect(wrapper.vm.computedPaddingLeft).toBe(42)
    expect(wrapper.vm.computedPaddingRight).toBe(84)
    wrapper.setProps({ app: true, clippedLeft: true, clippedRight: true })
    expect(wrapper.vm.computedPaddingLeft).toBe(0)
    expect(wrapper.vm.computedPaddingRight).toBe(0)
  })

  it('should calculate application top', async () => {
    const wrapper = mount(VToolbar, {
      propsData: {
        app: true,
        fixed: true,
        height: 21
      }
    })

    expect(wrapper.vm.$vuetify.application.top).toBe(21)
    wrapper.setProps({
      height: 42
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.$vuetify.application.top).toBe(42)

    wrapper.setProps({
      invertedScroll: true
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.$vuetify.application.top).toBe(0)
  })

  it('should properly calculate content height', () => {
    const wrapper = mount(VToolbar)

    wrapper.setProps({
      height: 999
    })
    expect(wrapper.vm.computedContentHeight).toBe(999)

    wrapper.setProps({
      height: null,
      dense: true
    })
    expect(wrapper.vm.computedContentHeight).toBe(wrapper.vm.heights.dense)

    wrapper.setProps({
      height: null,
      dense: false,
      prominent: true
    })
    expect(wrapper.vm.computedContentHeight).toBe(wrapper.vm.heights.desktop)

    wrapper.setProps({
      height: null,
      dense: false,
      prominent: false
    })
    Vue.set(wrapper.vm.$vuetify.breakpoint, 'width', 200)
    Vue.set(wrapper.vm.$vuetify.breakpoint, 'height', 100)
    expect(wrapper.vm.computedContentHeight).toBe(wrapper.vm.heights.mobileLandscape)
    Vue.set(wrapper.vm.$vuetify.breakpoint, 'width', 100)
    Vue.set(wrapper.vm.$vuetify.breakpoint, 'height', 200)
    expect(wrapper.vm.computedContentHeight).toBe(wrapper.vm.heights.mobile)
  })

  it('should set margin top', () => {
    const wrapper = mount(VToolbar, {
      propsData: {
        app: true
      }
    })

    Vue.set(wrapper.vm.$vuetify.application, 'bar', 24)
    expect(wrapper.vm.computedMarginTop).toBe(24)
  })

  it('should set active based on manual scroll', async () => {
    const wrapper = mount(VToolbar, {
      propsData: {
        scrollOffScreen: true
      }
    })

    expect(wrapper.vm.isActive).toBe(true)
    wrapper.setProps({ manualScroll: true })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.isActive).toBe(false)
  })

  it.skip('should set a custom target', async () => {
    const wrapper = mount(VToolbar, {
      propsData: {
        target: 'body'
      }
    })

    wrapper.vm.onScroll()
    expect(wrapper.vm.target).toBe('body')
  })

  it.skip('should set isScrollingUp', async () => {
    const wrapper = mount(VToolbar)

    await scrollWindow(100)
    expect(wrapper.vm.isScrollingUp).toBe(false)
    await scrollWindow(0)
    expect(wrapper.vm.isScrollingUp).toBe(true)
  })
})
