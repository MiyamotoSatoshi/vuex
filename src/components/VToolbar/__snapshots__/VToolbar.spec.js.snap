// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VToolbar.vue should render an extended toolbar 1`] = `

<nav class="toolbar toolbar--extended"
     style="margin-top: 0px; padding-right: 0px; padding-left: 0px;"
>
  <div class="toolbar__content"
       style="height: 56px;"
  >
  </div>
  <div class="toolbar__extension"
       style="height: 56px;"
  >
  </div>
</nav>

`;

exports[`VToolbar.vue should render an extended toolbar with specific height 1`] = `

<nav class="toolbar toolbar--extended"
     style="margin-top: 0px; padding-right: 0px; padding-left: 0px;"
>
  <div class="toolbar__content"
       style="height: 56px;"
  >
  </div>
  <div class="toolbar__extension"
       style="height: 42px;"
  >
  </div>
</nav>

`;
